src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -873,38 +873,28 @@ int MPIDI_OFI_flush_send_queue(void)
 {
     int mpi_errno = MPI_SUCCESS;
 
-    MPIDI_OFI_dynamic_process_request_t *reqs;
     /* TODO - Iterate over each NIC in addition to each VNI when multi-NIC within the same
      * process is implemented. */
     int num_vcis = MPIDI_OFI_global.num_vcis;
-    int num_reqs = num_vcis * 2;
-    reqs = MPL_malloc(sizeof(MPIDI_OFI_dynamic_process_request_t) * num_reqs, MPL_MEM_OTHER);
 
     /* Apparently by sending self messages can flush the send queue */
     int rank = MPIR_Process.rank;
     for (int vci = 0; vci < num_vcis; vci++) {
-        mpi_errno = flush_send(rank, 0, vci, &reqs[vci * 2]);
+        MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI_LOCK(vci));
```

Comment:
Is it correct to enter/exit the critical section and use `MPIR_ERR_CHECK` in the between the enter/exit calls?

Reply:
You are right. There is a risk of leak CS.